### ISSUE TITLE
Reopening Failed Sugar/Suite CRM +  Mautic Sync Issue - Found fix + Activity Patch download!  (Previously 7420/7138/7112)

### ISSUE THREAD

**Issue by cloudpotions** [mautic/mautic#9926]
Suite CRM: Version 7.11.5 / Sugar Version 6.5.25 (Build 344)
Mautic 3.3.2
Ionos Virtual server with Ubuntu 20.04 and Plesk 
PHP Version 7.3.27 - Fast CGI on both 

I can confirm that I have the 2 way sync working with Suite CRM and Mautic. I also installed on Suite CRM the "Web activity Module" which syncs over the web activities. After I Installed this + added the PROPER cron jobs (tested a bunch) - it syncs both ways perfectly. I will provide documentation below on this...

However, there is one major glitch that needs too be fixed... (and this is the main purpose of this post but I also wanted too describe what I did below too finally get all of this working)..... 

Situation - Mautic is only going to push/fetch  Leads where the date_modified is greater than the last_sync_date. And it literally set the last_sync_date on many of these records milliseconds after the cron jobs run... 

There is a hack though to fix this without code but it is resource intensive server side. What you can do is create a campaign that adds one point and then X minutes after minuses a point and put this campaign in a loop every X amount of minutes - Make sure the campaign contacts are allowed too "restart" the campaign radio button is selected. 

In a nutshell here is what I used too get my Sync working properly (I also synced with Callingly.com and now I have an automated call center solution for a newly built campaign). 

CRONS RUN from Bin/Console 

Mautic Crons 

mautic:integration:pushactivity --integration=Sugarcrm  (Web activity module)
mautic:integration:fetchleads --fetch-all --integration=Sugarcrm
mautic:integration:synccontacts --integration=Sugarcrm
cache:clear

+ I would suggest adding Segment Update, Campaign Update and Campaign Trigger as per documentation. 

Suite CRM  - run your cron.php file as suggested in documentation (Softaculous will set up cron automatically for you in Plesk)

** Web Activities Module can be [downloaded here
](https://drive.google.com/file/d/1LQaupM5znRPl0bZCnAv0uXP6U5dfzaUR/view?usp=sharing)

**Comment by npracht** [comment]
Hi @MyBpoConsultant !
Thanks a lot for your complete message but i don't understand the objective of it ?
Is there a bug ? Do you recommend an enhancement ?

**Comment by cloudpotions** [comment]
@npracht - Apologies for the late response. Yes there is a bug and I think it will fix other plugins too. Mautic is only going to push/fetch Leads where the date_modified is greater than the last_sync_date. And it literally sets the last_sync_date on many of these records milliseconds after the cron jobs run... Therefore they do not update... I tested this and got the sync too work on many leads after I changed a lead property and press save - those ones will sync but the others will not. One hack is to create a campaign that adds a point too all contacts every 15 minutes and then minuses a point but ideally the code in Mautic would be changed so one does not have to do this.
